llvm/include/llvm/ADT/STLExtras.h

```diff
@@ -1535,6 +1535,13 @@ constexpr decltype(auto) makeVisitor(CallableTs &&...Callables) {
   return detail::Visitor<CallableTs...>(std::forward<CallableTs>(Callables)...);
 }
 
+/// Backport of C++23 std::to_underlying.
+template <typename Enum>
+[[nodiscard]] constexpr typename std::underlying_type_t<Enum>
+to_underlying(Enum E) {
+  return static_cast<typename std::underlying_type_t<Enum>>(E);
```

Comment:
do we need the `typename` here?

Reply:
Apparently we don't need `typename` in both definition and return type: https://godbolt.org/z/nczqn85sn
I'll remove it

Reply:
Done